server/game/core/ability/CardAbilityStep.js

```diff
@@ -135,7 +135,7 @@ class CardAbilityStep extends PlayerOrCardAbility {
         let effectShouldResolve;
 
         if (this.properties.ifYouDo) {
-            ifAbility = this.properties.ifYouDo;
+            ifAbility = typeof this.properties.ifYouDo === 'function' ? this.properties.ifYouDo(context) : this.properties.ifYouDo;
```

Comment:
I think this change isn't needed anymore due to the fix we did on line 147 in a recent merge, can just do `ifAbility = this.properties.ifYouDo;`